process/serialize.go

```diff
@@ -0,0 +1,237 @@
+package process
+
+import (
+	"sort"
+	"strconv"
+
+	"github.com/mesg-foundation/engine/hash/hashserializer"
+)
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("2", data.Name)
+	ser.Add("4", processNodes(data.Nodes))
+	ser.Add("5", processEdges(data.Edges))
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("1", data.Key)
+	ser.Add("2", data.GetResult())
+	ser.Add("3", data.GetEvent())
+	ser.Add("4", data.GetTask())
+	ser.Add("5", data.GetMap())
+	ser.Add("6", data.GetFilter())
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Result) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("2", data.InstanceHash.String())
+	ser.AddString("3", data.TaskKey)
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Event) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("2", data.InstanceHash.String())
+	ser.AddString("3", data.EventKey)
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Task) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("2", data.InstanceHash.String())
+	ser.AddString("3", data.TaskKey)
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Map) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.Add("1", mapProcessNodeMapOutput(data.Outputs))
+	return ser.HashSerialize()
+}
+
+type mapProcessNodeMapOutput map[string]*Process_Node_Map_Output
+
+// HashSerialize returns the hashserialized string of this type
+func (data mapProcessNodeMapOutput) HashSerialize() string {
+	ser := hashserializer.New()
+	keys := make([]string, 0, len(data))
+	for k := range data {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		ser.Add(key, data[key])
+	}
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Map_Output) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("2", data.GetStringConst())
```

Comment:
Could we add the null just to make sure we don't forget even it will be empty? That will reduce confusion.